# CREDIT LOAN RISK DATA ANALYSIS

Credits to Lao Tse for the dataset.
You can access the dataset [here](https://www.kaggle.com/datasets/laotse/credit-risk-dataset)

### Introduction

In this Jupyter notebook, the analyst used the Python programming language to analyze the following dataset about the credit risk of borrowers involved, with the intention to examine and analyze trends, findings, and what recommendations can be given based on the results.

---
### The Data Collection

The dataset consists of the said columns and their values collected from Lao Tse, a Kaggle dataset provider. The following data are presented using Python with Pandas, Matplotlib, and Numpy libraries to enable further control and examination of the values. The dataset is stored in a CSV file, and is imported in the notebook using tha Pandas library. 

The following table showcases the columns involved and their descriptions.

| Column Name | Description |
| --- | --- |
| person_age | Age of the borrower |
| person_income | Annual income |
| person_home_ownership | Status of home ownership |
| person_emp_length | Employment length in terms of years |
| loan_intent | Reason of loaning |
| loan_grade | Grade associated with the borrower or loan |
| loan_amnt | Amount of money loaned |
| loan_int_rate | Interest rate |
| loan_status | Status of the loan. ( 1 means the loan is still active) |
| loan_percent_income | Percentage of income that goes into the loan. |
| cb_person_default_on_file | If the borrower has failed to make scheduled payments|
| cb_person_cred_hist_length | Length of credit history in years |

---
### Data Cleaning and Preparation

The provided dataset is divided into separate columns, each consisting of string and numerical values. To further analyze the dataset, the analyst has gather more information using python. A summary table has also been provided.

| Column Name | Actions |
| --- | --- |
| person_age | Identified and cleaned values with invalid maximum age. Replace with more viable values |
| person_income |  |
| person_home_ownership |  |
| person_emp_length |  |
| loan_intent |  |
| loan_grade |  |
| loan_amnt |  |
| loan_int_rate |  |
| loan_status |  |
| loan_percent_income | . |
| cb_person_default_on_file |  |
| cb_person_cred_hist_length |  |

The analyst imports the dataset and used the Pandas function of `.info()`.

In [193]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sb

dataset = pd.read_csv('credit_risk_dataset.csv')
dataset.info()

<class 'pandas.DataFrame'>
RangeIndex: 32581 entries, 0 to 32580
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   person_age                  32581 non-null  int64  
 1   person_income               32581 non-null  int64  
 2   person_home_ownership       32581 non-null  str    
 3   person_emp_length           31686 non-null  float64
 4   loan_intent                 32581 non-null  str    
 5   loan_grade                  32581 non-null  str    
 6   loan_amnt                   32581 non-null  int64  
 7   loan_int_rate               29465 non-null  float64
 8   loan_status                 32581 non-null  int64  
 9   loan_percent_income         32581 non-null  float64
 10  cb_person_default_on_file   32581 non-null  str    
 11  cb_person_cred_hist_length  32581 non-null  int64  
dtypes: float64(3), int64(5), str(4)
memory usage: 3.0 MB


The data above shows the general information regarding the data. It consists of 32581 entries for 12 columns. According tot he non-null count, there are some values that are null in `person_emp_length`, and `loan_int_rate`. Null means that there are no values given and does not mean that the value is 0. The data types consists of three types, int64, str, and float64. `int64` refers to whole numbers, whether negative or positive like -1, 4, 0, 99. `str` refers to values that are stored as text like RENT or MORTGAGE. `float64` refers to numbers with decimal places unlike `int64`, like 1.32, 0.00, or -13.23.

In [194]:
dataset.describe()

,person_age,person_income,person_emp_length,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_cred_hist_length
count,32581.000000,3.258100e+04,31686.000000,32581.000000,29465.000000,32581.000000,32581.000000,32581.000000
mean,27.734600,6.607485e+04,4.789686,9589.371106,11.011695,0.218164,0.170203,5.804211
std,6.348078,6.198312e+04,4.142630,6322.086646,3.240459,0.413006,0.106782,4.055001
min,20.000000,4.000000e+03,0.000000,500.000000,5.420000,0.000000,0.000000,2.000000
25%,23.000000,3.850000e+04,2.000000,5000.000000,7.900000,0.000000,0.090000,3.000000
50%,26.000000,5.500000e+04,4.000000,8000.000000,10.990000,0.000000,0.150000,4.000000
75%,30.000000,7.920000e+04,7.000000,12200.000000,13.470000,0.000000,0.230000,8.000000
max,144.000000,6.000000e+06,123.000000,35000.000000,23.220000,1.000000,0.830000,30.000000


By using `.describe()`, the dataset's summary of numerical values can be seen. This includes the number of enties, mean, standard deviation, minimum values, quartiles, and the maximum numerical values. The summary provides values that are essential in data analysis.

##### Age

This column represents the ages of the borrowers when the dataset was recorded. Analyzing this column can provide insights regarding the ages and other data. To further examine the age of the borrowers, the analyst have used the following code that targets the age of the borrowers specifically by assigning the column into a separate series of data:

In [195]:
person_ages = dataset['person_age']
person_ages.info()

<class 'pandas.Series'>
RangeIndex: 32581 entries, 0 to 32580
Series name: person_age
Non-Null Count  Dtype
--------------  -----
32581 non-null  int64
dtypes: int64(1)
memory usage: 254.7 KB


The result shows the data as a series of integers consisting of 32,581 entries.

In [196]:
person_ages.describe()

count    32581.000000
mean        27.734600
std          6.348078
min         20.000000
25%         23.000000
50%         26.000000
75%         30.000000
max        144.000000
Name: person_age, dtype: float64

Using `.describe()` shows the statistical summary of the column. According to the function results, the maximum value is `144` which should be impossible for a human lifespan and treated as a data entry error. To further analyze the concern, the analyst finds the ages equivalent to `144` and the maximum age lower than the value.

In [197]:
person_ages[person_ages != 144].max()

np.int64(123)

Excluding `144`, another issue has occured in which the new maximum value that is not likely to occur considering human lifespan, there is also a chance that this is an entry error. That new number is `123` which is close to the oldest age a person has achieved which is 122. To analyze this further, let us examine the remaining data without the values of `123` and `144`.

In [198]:
person_ages[(person_ages != 144) & (person_ages != 123)].max()

np.int64(94)

Since the new possible maximum age is `94`, there is a very high chance that this is the new maximum value. This is because `94` is a valid age to get a loan like [PVAO Pension Loans](https://www.veteransbank.com.ph/pension-loan) which allows citizens 65 years old and above to qualify for a row. Now that we have a new valid maximum age, let us deal with the outliers by identifying their other details.

In [199]:
filtered_ages = dataset[['person_age','person_income']]
filtered_ages[ (filtered_ages['person_age'] == 144) | ( filtered_ages['person_age']  == 123)]


,person_age,person_income
81,144,250000
183,144,200000
575,123,80004
747,123,78000
32297,144,6000000


In [200]:
person_ages[person_ages.eq(123)].count()

np.int64(2)

In [201]:
person_ages[person_ages.eq(144)].count()

np.int64(3)

The results indicate that the rows with `144` as the age only amount up to three rows:`81`, `183`, `32297`. On the other hand, only two rows with the value of `123` are found: `575`, `747`.

To treat values with `144`, the analyst have considered the following options as the new value:

> `null`: the analyst have considered to use null as the replacement value in order to not skew the statistics but having considered that all the columns are needed to be analyzed carefully, `null` will not be a suitable value in this case.

> `14`: Considering that the age is too low for a loan, and will standout as the only value lower then the minimum value found, `14` is not a possible value. The income also cannot be viable if the person is 14.

> `44`: The only possible value because it is in the range between minimum and maximum values and at the range of ages where loans are still applicable to apply.

To treat values with `123`, the analyst have considered the following options as the new value:

> `null`: the analyst have also considered to use null as the replacement value in order to not skew the statistics but having considered that all the columns are needed to be analyzed carefully, `null` will not be a suitable value in this case.

> `12 or 13`: These ages also considered too low for a loan, therefore not a suitable value. The income also cannot be viable if the person is 12 or 13.

> `21 or 31`: Also possible but very unlikely since the keyboard layout which starts from 1 then 2 and 3. If it is `21` or `31`, we are more likely to see `121` or `231`.

> `23`: The only possible value because it is also in the range between minimum and maximum values and at the range of ages where loans are still applicable to apply.

Based on the consideration and data review, it is more applicable to replace `144` with `44` and `123` with `23`. In order to change the values, the following code is used:

In [202]:
age_with_144 = filtered_ages['person_age'] == 144
filtered_ages.loc[age_with_144, 'person_age'] = 44
filtered_ages.iloc[[81,183,32297]]

,person_age,person_income
81,44,250000
183,44,200000
32297,44,6000000


In [203]:
age_with_144 = filtered_ages['person_age'] == 123
filtered_ages.loc[age_with_144, 'person_age'] = 23
filtered_ages.iloc[[575,747]]

,person_age,person_income
575,23,80004
747,23,78000


During the process, the analyst stored the rows with the age values of `123` and `144` and change the value of `'person_age'` for the said rows to match the chosen value which is `23` for `123` and `44` for `144`. Then the same rows were examined again for their new values and based on the result, the ages has changed, therefore it can be concluded that the column has been cleaned and the same changes can now be used on the main copy of the dataset with cleaned data. The cleaned column is also stored as a separate CSV file in the `cleaned_data` folder for backup. This can be done using the following code:

In [204]:
cleaned_dataset = dataset.copy()
cleaned_dataset = cleaned_dataset.drop(columns=['person_age'])
cleaned_dataset['Clean_person_age'] = filtered_ages['person_age']

cleaned_dataset['Clean_person_age'].to_csv('cleaned_data/cleaned_person_ages.csv', index=False)

cleaned_dataset

,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length,Clean_person_age
0,59000,RENT,123.0,PERSONAL,D,35000,16.02,1,0.59,Y,3,22
1,9600,OWN,5.0,EDUCATION,B,1000,11.14,0,0.10,N,2,21
2,9600,MORTGAGE,1.0,MEDICAL,C,5500,12.87,1,0.57,N,3,25
3,65500,RENT,4.0,MEDICAL,C,35000,15.23,1,0.53,N,2,23
4,54400,RENT,8.0,MEDICAL,C,35000,14.27,1,0.55,Y,4,24
...,...,...,...,...,...,...,...,...,...,...,...,...
32576,53000,MORTGAGE,1.0,PERSONAL,C,5800,13.16,0,0.11,N,30,57
32577,120000,MORTGAGE,4.0,PERSONAL,A,17625,7.49,0,0.15,N,19,54
32578,76000,RENT,3.0,HOMEIMPROVEMENT,B,35000,10.99,1,0.46,N,28,65
32579,150000,MORTGAGE,5.0,PERSONAL,B,15000,11.48,0,0.10,N,26,56


In [205]:
cleaned_dataset.iloc[[81,183,32297]]

,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length,Clean_person_age
81,250000,RENT,4.0,VENTURE,C,4800,13.57,0,0.02,N,3,44
183,200000,MORTGAGE,4.0,EDUCATION,B,6000,11.86,0,0.03,N,2,44
32297,6000000,MORTGAGE,12.0,PERSONAL,C,5000,12.73,0,0.00,N,25,44


In [206]:
cleaned_dataset.iloc[[575,747]]

,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length,Clean_person_age
575,80004,RENT,2.0,EDUCATION,B,20400,10.25,0,0.25,N,3,23
747,78000,RENT,7.0,VENTURE,B,20000,NaN,0,0.26,N,4,23


To wrap things up, the older column was replaced dropped and new values for their ages are not stored. The filtered column is added at the end of the table. When we take a look back to the columns with dirty values, they are now cleaned.

#### Currency

Before we head to the next column, the analyst have noticed that no currency was used. Therefore, Dollars ($) will be used on the data analysis as a stand-in for a real currency.


##### Income

This column stores the data regarding the yearly income of the borrowers when the dataset was made. Income can give many insights and is one of the most important variable when considering to provide a loan. The analyst used the following code to further examine the column's data.

In [207]:
person_incomes = dataset['person_income']
person_incomes

0         59000
1          9600
2          9600
3         65500
4         54400
          ...  
32576     53000
32577    120000
32578     76000
32579    150000
32580     42000
Name: person_income, Length: 32581, dtype: int64

In [208]:
person_incomes.info()

<class 'pandas.Series'>
RangeIndex: 32581 entries, 0 to 32580
Series name: person_income
Non-Null Count  Dtype
--------------  -----
32581 non-null  int64
dtypes: int64(1)
memory usage: 254.7 KB


In [209]:
person_incomes.describe().apply(lambda x: format(x, 'f'))

count      32581.000000
mean       66074.848470
std        61983.119168
min         4000.000000
25%        38500.000000
50%        55000.000000
75%        79200.000000
max      6000000.000000
Name: person_income, dtype: str

The python code resulted in more detailed information regarding the column. From the data, it consists of 32581 entries and as intergers with no null values. The numerical summaries are also presented like the count, mean, standard deviation, minimum and maximum value, quartiles, and maximum values. One outlier that pops up is the maximum income which amounts to `6,000,000` and compared to the third quartile, a considerable jump in income has been observed. To further analyze the high amount of incomes, the analyst examined all the income between the third quartile and maximum income using the following code:

In [210]:
dataset[dataset['person_income'] == 6000000]

,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
32297,144,6000000,MORTGAGE,12.0,PERSONAL,C,5000,12.73,0,0.0,N,25


In [211]:
third_quartile_incomes = person_incomes[(person_incomes >= person_incomes.quantile(0.75)) & (person_incomes < 6000000) ]
third_quartile_incomes.describe().apply(lambda x: format(x,'f'))

count       8157.000000
mean      121808.686159
std        77703.101556
min        79200.000000
25%        89532.000000
50%       102000.000000
75%       128000.000000
max      2039784.000000
Name: person_income, dtype: str

After gathering all the values of between the third quartile and the `6,000,000` income, the new maximum income is `2039784`, which means that there is a huge gap between the largest income and the second most largest income. This is important to know as now we can treat `6,000,000` as the only outlier in its range. Therefore, it can be surmised that there is one value in the dataset in this column that can possibly an entry error. In this case, analyzing the rows of income with the same range values as the outlier can provide insights. The code below was used:

In [212]:

same_home_ownership=(dataset['person_home_ownership']) == 'MORTGAGE'
same_emp_length = (dataset['person_emp_length']) == 12.0
same_loan_intent = (dataset['loan_intent']) == 'PERSONAL'
same_loan_grade = (dataset['loan_grade']) == 'C'
same_loan_amount = (dataset['loan_amnt']) == '5000'
same_loan_int_rate = (dataset['loan_int_rate']) == 12.73
same_loan_status = (dataset['loan_status']) == '0'
same_loan_percent_income = (dataset['loan_percent_income']) == 0.0
same_default = (dataset['cb_person_default_on_file']) == 'N'
same_cred_hist_length = (dataset['loan_grade']) == 25

same_age_range =(dataset['person_age'] == 144) | (dataset['person_age'] == 44 )

x = dataset[(
  (same_age_range) |
  (same_home_ownership) |
  (same_emp_length) |
  (same_loan_intent) |
  (same_loan_grade) |
  (same_loan_amount) |
  (same_loan_int_rate) |
  (same_loan_status) |
  (same_loan_percent_income) |
  (same_default) |
  (same_cred_hist_length)
  ) 
  & (dataset['person_income'] != 6000000 )]
x['person_income'].describe().apply(lambda x: format(x,'f'))

count      31254.000000
mean       66342.287547
std        52867.262757
min         4000.000000
25%        39000.000000
50%        55680.000000
75%        80000.000000
max      2039784.000000
Name: person_income, dtype: str

Excluding the outlier, the income of borrowers in similar categories are relatively consistent. Therefore, the analyst concludes that the income of `6,000,000` is impossible and an entry error due to having an extra zero at the end. The analyst then considered the new value to be `60,000` removing a trailing `0` as after analyzing, entries at similar categories are closer to that value of the mean, making `60,000` a viable replacement. The table below shows the entries summary:

In [213]:
dataset[(dataset['person_income'] != 6000000) & (((dataset['person_age'] >= 40) & dataset['person_age'] <= 50) & (dataset['loan_grade'] == 'C'))].describe().map('{:,.2f}'.format)

,person_age,person_income,person_emp_length,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_cred_hist_length
count,"6,457.00","6,457.00","6,320.00","6,457.00","5,827.00","6,457.00","6,457.00","6,457.00"
mean,27.78,"64,002.77",4.45,"9,214.52",13.46,0.21,0.17,5.86
std,6.40,"57,548.89",3.88,"6,044.84",0.96,0.41,0.11,4.08
min,20.00,"4,000.00",0.00,500.00,6.00,0.00,0.00,2.00
25%,23.00,"36,000.00",2.00,"4,900.00",12.98,0.00,0.09,3.00
50%,26.00,"53,000.00",4.00,"8,000.00",13.48,0.00,0.15,4.00
75%,30.00,"76,000.00",7.00,"12,000.00",13.99,0.00,0.23,8.00
max,144.00,"2,039,784.00",29.00,"35,000.00",16.11,1.00,0.71,30.00


The column can now be cleaned of dirty values and exported as its own file as backup. The old value was replaced using the following command:

In [214]:
person_incomes.iloc[32297]

np.int64(6000000)

In [215]:
entry_error = dataset['person_income'] == 6000000
filtered_incomes = dataset[['person_income','person_age']]
filtered_incomes.loc[entry_error, 'person_income'] = 60000
filtered_incomes.iloc[32297]


person_income    60000
person_age         144
Name: 32297, dtype: int64

In [216]:
cleaned_dataset = cleaned_dataset.drop(columns=['person_income'])
cleaned_dataset['Clean_person_income'] = filtered_incomes['person_income']
cleaned_dataset

,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length,Clean_person_age,Clean_person_income
0,RENT,123.0,PERSONAL,D,35000,16.02,1,0.59,Y,3,22,59000
1,OWN,5.0,EDUCATION,B,1000,11.14,0,0.10,N,2,21,9600
2,MORTGAGE,1.0,MEDICAL,C,5500,12.87,1,0.57,N,3,25,9600
3,RENT,4.0,MEDICAL,C,35000,15.23,1,0.53,N,2,23,65500
4,RENT,8.0,MEDICAL,C,35000,14.27,1,0.55,Y,4,24,54400
...,...,...,...,...,...,...,...,...,...,...,...,...
32576,MORTGAGE,1.0,PERSONAL,C,5800,13.16,0,0.11,N,30,57,53000
32577,MORTGAGE,4.0,PERSONAL,A,17625,7.49,0,0.15,N,19,54,120000
32578,RENT,3.0,HOMEIMPROVEMENT,B,35000,10.99,1,0.46,N,28,65,76000
32579,MORTGAGE,5.0,PERSONAL,B,15000,11.48,0,0.10,N,26,56,150000


In [217]:
cleaned_dataset['Clean_person_income'].to_csv('cleaned_data/cleaned_person_incomes.csv',index=False)
person_incomes.describe().apply(lambda x: format(x, 'f'))

count      32581.000000
mean       66074.848470
std        61983.119168
min         4000.000000
25%        38500.000000
50%        55000.000000
75%        79200.000000
max      6000000.000000
Name: person_income, dtype: str

The value has been replaced and as we can clearly see, the new maximum value is no longer `6,000,000`. The cleaned column now has a backup CSV file, and on the clean dataset, old column is now dropped, replaced with the cleaned column for `person_income`.

##### Home Ownership

In this column, the collected data represents the type of home ownership borrowers have. Analyzing this column is importany as it prove to be useful in identifying trends or insights regarding their loans. To further analyze the column, the analyst used the following code:

In [218]:
home_ownerships = dataset['person_home_ownership']
home_ownerships.info()

<class 'pandas.Series'>
RangeIndex: 32581 entries, 0 to 32580
Series name: person_home_ownership
Non-Null Count  Dtype
--------------  -----
32581 non-null  str  
dtypes: str(1)
memory usage: 254.7 KB


In [219]:
home_ownerships.describe()

count     32581
unique        4
top        RENT
freq      16446
Name: person_home_ownership, dtype: object

Based on the results, the columns contains string of for different types with 32581 entries. The most frequent home ownership is `RENT` with a total number of `16,446`. To identify the other kinds of ownership, the analyst used the following:

In [220]:
home_ownerships.drop_duplicates(keep='last')

32291       OTHER
32572         OWN
32579    MORTGAGE
32580        RENT
Name: person_home_ownership, dtype: str

There are four types, `OTHER`, `OWN`, `MORTGAGE`, and `RENT`. Other than this, the table has no concerns or data to examine, thus the data can be considered clean. The next step follows the analyst creating a new CSV file for the column and updating the old dataset.

In [ ]:
cleaned_dataset = cleaned_dataset.drop(columns=['person_home_ownership'])
cleaned_dataset['Cleaned_person_home_ownership'] = home_ownerships
cleaned_dataset['Cleaned_person_home_ownership'].to_csv('cleaned_data/cleaned_person_home_ownerships.csv', index=False)

cleaned_dataset

,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length,Clean_person_age,Clean_person_income,Cleaned_person_home_ownership
0,123.0,PERSONAL,D,35000,16.02,1,0.59,Y,3,22,59000,RENT
1,5.0,EDUCATION,B,1000,11.14,0,0.10,N,2,21,9600,OWN
2,1.0,MEDICAL,C,5500,12.87,1,0.57,N,3,25,9600,MORTGAGE
3,4.0,MEDICAL,C,35000,15.23,1,0.53,N,2,23,65500,RENT
4,8.0,MEDICAL,C,35000,14.27,1,0.55,Y,4,24,54400,RENT
...,...,...,...,...,...,...,...,...,...,...,...,...
32576,1.0,PERSONAL,C,5800,13.16,0,0.11,N,30,57,53000,MORTGAGE
32577,4.0,PERSONAL,A,17625,7.49,0,0.15,N,19,54,120000,MORTGAGE
32578,3.0,HOMEIMPROVEMENT,B,35000,10.99,1,0.46,N,28,65,76000,RENT
32579,5.0,PERSONAL,B,15000,11.48,0,0.10,N,26,56,150000,MORTGAGE


##### Employment Length

This column represents the borrower's employment length in terms of years. The column can give insights about the borrower and information about them which can be helpful to find relationships regarding borrower information and their loan. The analyst used the following code as the next step for the analysis:

In [222]:
employment_length = dataset['person_emp_length']
employment_length.info()

<class 'pandas.Series'>
RangeIndex: 32581 entries, 0 to 32580
Series name: person_emp_length
Non-Null Count  Dtype  
--------------  -----  
31686 non-null  float64
dtypes: float64(1)
memory usage: 254.7 KB


In [223]:
employment_length.describe()

count    31686.000000
mean         4.789686
std          4.142630
min          0.000000
25%          2.000000
50%          4.000000
75%          7.000000
max        123.000000
Name: person_emp_length, dtype: float64

In [ ]:
dataset[dataset['person_emp_length'].isnull(), ['person_emp_length','loan_percent_income']]

InvalidIndexError: (0        False
1        False
2        False
3        False
4        False
         ...  
32576    False
32577    False
32578    False
32579    False
32580    False
Name: person_emp_length, Length: 32581, dtype: bool, ['person_emp_length', 'loan_percent_income'])

#### Loan Grade

There are many ways a loan can be graded. In the dataset, it uses an alphabetical grading, ranging from A to F based on the propriety risk rating scale. The details of the loan grades is shown below:

| Grade | Meaning |
| --- | --- |
| A | Lowest risk |
| B | Low risk |
| C | Average risk |
| D | Moderate risk |
| E | High risk |
| F | Very High risk |

The grades determine the 'risk' or the frequency of the debter to fall into a historical default. Therefore borrowers tend to charge high interest rate to compensate for the possible risk.